plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -79,6 +79,8 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
     }
 
     override fun embedType(type: ConeKotlinType): TypeEmbedding = when {
+        type is ConeErrorType -> throw IllegalArgumentException("Encountered an erroneous type: $type")
+        type is ConeTypeParameterType -> NullableTypeEmbedding(AnyTypeEmbedding)
```

Comment:
I see a potential issue here, though I don't know whether it can happen in practice.  Let's say you have a `fun <T, S> f(t: T, s: S)`.  Can you at any point do `t = s`, or anything to that effect?  Since our typecases would assume that these are the same type, when in fact they aren't.  Or would this not matter anyway, since the typecast can't do anything interesting?

Reply:
This doesn't matter, for one, the Kotlin type systems disallows such assignments and the only semantically interesting thing our types can do is an `is` check. But `is T` is not allows unless `T` is reified which I haven't looked at yet.